drivers/si114x/si114x_saul.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2016-2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_si114x
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for Si114x devices family
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *              Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#include "saul.h"
+#include "si114x.h"
+#include "xtimer.h"
+
+static int read_uv(const void *dev, phydat_t *res)
+{
+    si114x_t *d = (si114x_t *)dev;
+
+    res->val[0] = si114x_read_uv(d);
+    res->unit = UNIT_NONE;
+    res->scale = 0;
+    return 1;
+}
+
+static int read_ir(const void *dev, phydat_t *res)
+{
+    si114x_t *d = (si114x_t *)dev;
+
+    res->val[0] = si114x_read_ir(d);
+    res->unit = UNIT_LUX;
+    res->scale = 0;
+    return 1;
+}
+
+static int read_visible(const void *dev, phydat_t *res)
+{
+    si114x_t *d = (si114x_t *)dev;
+
+    res->val[0] = si114x_read_visible(d);
+    res->unit = UNIT_LUX;
+    res->scale = 0;
+    return 1;
+}
+
+static int read_distance(const void *dev, phydat_t *res)
+{
+    si114x_t *d = (si114x_t *)dev;
+
+    res->val[0] = si114x_read_distance(d);
+    res->unit = UNIT_M;
```

Comment:
@basilfx, so it should be `UNIT_NONE` here ?

Reply:
Yes.

The datasheet doesn't mention this explicitly, but [AN498](https://www.silabs.com/documents/public/application-notes/AN498.pdf) does (page 50).

Reply:
Changed this and also adapted test program to mention `distance [cnt]` instead of `distance [m]`.

Reply:
And in the driver header file.